File: spring/yt-developeryumi-springboot-tdd/secondtest/src/main/java/com/example/secondtest/api/PageController.java
package com.example.secondtest.api;

import com.example.secondtest.domain.post.dto.PostRequestDTO;
import com.example.secondtest.domain.post.entity.PostEntity;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;

import java.util.ArrayList;
import java.util.List;

@Controller
public class PageController {

    @GetMapping("/page")
    public String getPage(Model model) {

        List<PostEntity> postEntityList = new ArrayList<>();
        model.addAttribute("POSTLIST", postEntityList);

        return "page";
    }

    @PostMapping("/page")
    public String postPage(PostRequestDTO dto) {

        return "redirect:/page";
    }

}

File: spring/inflearn-steven-kotlin-springboot-webmvc/mvc/src/main/kotlin/com/example/mvc/controller/page/PageController.kt
package com.example.mvc.controller.page

import com.example.mvc.model.http.UserRequest
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.ResponseBody

@Controller
class PageController {

    @GetMapping("/main")
    fun main(): String {
        println("init main")
        return "main.html"
    }

    @ResponseBody
    @GetMapping("/test")
    fun response(): UserRequest {
        return UserRequest().apply {
            this.name = "steve"
        }
    }
}